pyincore/models/networkdataset.py

```diff
@@ -173,12 +173,17 @@ def get_links(self):
     def get_graph(self):
         return self.graph.get_csv_reader()
 
-    def get_graph_networkx(self, from_node_fld="fromnode", to_node_fld="tonode", directed=False):
+    def get_graph_networkx(self, from_node_fld="fromnode", to_node_fld="tonode", directed=False, numeric=True):
         if directed:
             G = nx.DiGraph()
         else:
             G = nx.Graph()
         for edge in list(self.get_graph()):
-            G.add_edge(edge[from_node_fld], edge[to_node_fld])
-
-        return G
\ No newline at end of file
+            # the csv_reader in the get_graph turns integer into string; sometimes the network needs to be
+            # constructed with integer as node id
+            if numeric:
```

Comment:
you might want to try & except to catch the error when converting to int() doesn't work? you can also ignore this one

Reply:
I can the default valueError is informational enough for except
```
 int('a')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'a'
```